embassy-rp/src/spi.rs

```diff
@@ -57,8 +64,8 @@ fn calc_prescs(freq: u32) -> (u8, u8) {
     ((presc * 2) as u8, (postdiv - 1) as u8)
 }
 
-impl<'d, T: Instance> Spi<'d, T> {
-    pub fn new(
+impl<'d, T: Instance, M: Mode> Spi<'d, T, M> {
+    pub fn new_blocking(
```

Comment:
to be consistent, `new`s need some rearranging:

```rust
impl<'d, T: Instance> Spi<'d, T, Blocking> {
    fn new_blocking
    fn new_blocking_rxonly
    fn new_blocking_txonly
}
impl<'d, T: Instance> Spi<'d, T, Async> {
    fn new
    fn new_rxonly
    fn new_txonly
}
```

Blocking constructors are inside an `impl` block that fixes `M=Blocking`, same for async ones. This ensures you can't construct an async SPI with the blocking constructor or vice versa.